### PR TITLE
SUPP0RT-1293: Allowed composite elements in Maestro notification reci…

### PR DESCRIPTION
**Pull request by jekuaitk**
…pient

https://jira.itkdev.dk/browse/SUPP0RT-1293

### CODE REVIEW

File: modules/os2forms_forloeb/src/MaestroHelper.php [hunk 1]
```diff
@@ -466,6 +466,21 @@ public function renderNotification(WebformSubmissionInterface $submission, strin
...
+    // Handle composite elements.
+    if ($recipient === NULL) {
+      if (str_contains($recipientElement, '__')) {
+        $keys = explode('__', $recipientElement);
+        $result = $data;
+        foreach ($keys as $key) {
+          if (array_key_exists($key, $result)) {
+            $result = $result[$key];
+          }
+        }
```
Comment (rimi-itk) [inline modules/os2forms_forloeb/src/MaestroHelper.php:478]: https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Component%21Utility%21NestedArray.php/function/NestedArray%3A%3AgetValue/10 is your friend.

File: modules/os2forms_forloeb/src/Plugin/WebformHandler/MaestroNotificationHandler.php [hunk 2]
```diff
@@ -240,6 +240,28 @@ private function getRecipientElements(): array {
       'cvr_value_element',
       'os2forms_person_lookup',
     ];
+
+    // Expand composite elements, NOT custom composite elements.
+    foreach ($elements as $key => $element) {
+      $orgElement = $this->getWebform()->getElement($key);
```
Comment (rimi-itk) [inline modules/os2forms_forloeb/src/Plugin/WebformHandler/MaestroNotificationHandler.php:246]: What does “org” mean here? “Original”? “Organ”? “Organization”?

```suggestion
      $formElement = $this->getWebform()->getElement($key);
```

File: modules/os2forms_forloeb/src/Plugin/WebformHandler/MaestroNotificationHandler.php [hunk 3]
```diff
@@ -240,6 +240,28 @@ private function getRecipientElements(): array {
...
+
+      if (array_key_exists('#webform_composite_elements', $orgElement)) {
+        foreach ($orgElement['#webform_composite_elements'] as $compositeElement) {
+          // If composite element is not accessible ignore it.
+          if (array_key_exists('#access', $compositeElement) && !$compositeElement['#access']) {
```
Comment (rimi-itk) [inline modules/os2forms_forloeb/src/Plugin/WebformHandler/MaestroNotificationHandler.php:251]:
```suggestion
          if (!($compositeElement['#access'] ?? TRUE)) {
```

File: modules/os2forms_forloeb/src/MaestroHelper.php [hunk 4]
```diff
@@ -466,6 +466,21 @@ public function renderNotification(WebformSubmissionInterface $submission, strin
       ?? $data[$recipientElement]
       ?? NULL;
 
+    // Handle composite elements.
+    if ($recipient === NULL) {
+      if (str_contains($recipientElement, '__')) {
```
[the remark this answers is not in the document]
Comment (jekuaitk) [reply modules/os2forms_forloeb/src/MaestroHelper.php:471]: Added comment explaining that it is how keys of composite subelements are generated.

File: modules/os2forms_forloeb/src/MaestroHelper.php (hunk 1 above)
Comment (jekuaitk) [reply modules/os2forms_forloeb/src/MaestroHelper.php:478]: Used `NestedArray`.

File: modules/os2forms_forloeb/src/Plugin/WebformHandler/MaestroNotificationHandler.php (hunk 2 above)
Comment (jekuaitk) [reply modules/os2forms_forloeb/src/Plugin/WebformHandler/MaestroNotificationHandler.php:246]: Renamed.

File: modules/os2forms_forloeb/src/Plugin/WebformHandler/MaestroNotificationHandler.php (hunk 3 above)
Comment (jekuaitk) [reply modules/os2forms_forloeb/src/Plugin/WebformHandler/MaestroNotificationHandler.php]: Nice, fixed.

**Review by rimi-itk — APPROVED** [review]
Looks good.

**Review by rimi-itk — APPROVED** [review]

**Pull request merged**